src/audio/kpb.c

```diff
@@ -337,6 +337,11 @@ static int kpb_prepare(struct comp_dev *dev)
 
 	trace_kpb("kpb_prepare()");
 
+	if (kpb->state == KPB_STATE_RESETTING ||
+	    kpb->state == KPB_STATE_RESET_FINISHING) {
+		return -EBUSY;
+	}
```

Comment:
Probably worth a trace message here for debugability.

Reply:
Definitely.